focus/internals/src/lib/model/selection/operations.rs

```diff
@@ -7,8 +7,24 @@ use super::*;
 
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 pub enum OperationAction {
-    Add,
-    Remove,
+    Add(AddOptions),
+    Remove(RemoveOptions),
+}
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct AddOptions {
+    pub unroll: bool,
+}
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct RemoveOptions {
+    pub all: bool,
+}
+
+pub fn default_add() -> OperationAction {
```

Comment:
Could you implement these as static functions on the enum?

impl OperationAction { pub fn default_add() -> OperationAction { ... } }

And then call with OperationAction::default_add()
Or something to that extent. I just think that when seeing a standing "default_add" in the code doesn't convey enough information about what type of default it is.